src/Carnac.Logic/KeyMonitor/DesktopLockEventService.cs

```diff
@@ -1,35 +1,25 @@
 using System;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
 using Microsoft.Win32;
 
 namespace Carnac.Logic.KeyMonitor
 {
     public class DesktopLockEventService : IDesktopLockEventService
     {
-        public event EventHandler<EventArgs> DesktopUnlockedEvent;
-        public event EventHandler<EventArgs> DesktopLockedEvent;
-
-        public DesktopLockEventService()
+        public IObservable<SessionSwitchEventArgs> GetSessionSwitchStream()
         {
-            SystemEvents.SessionSwitch += OnSystemEventsOnSessionSwitch;
-        }
-
-        void OnSystemEventsOnSessionSwitch(object sender, SessionSwitchEventArgs args)
-        {
-            if (args.Reason == SessionSwitchReason.SessionUnlock)
+            // Cannot use Observable.FromEventPattern as it causes an exception about security or something
```

Comment:
error = {"Cannot bind to the target method because its signature or security transparency is not compatible with that of the delegate type."}

Managed to get it working with:

``` csharp
return Observable.FromEvent<SessionSwitchEventHandler, SessionSwitchEventArgs>(
                handler => (sender, e) => handler(e),
                add => SystemEvents.SessionSwitch += add, 
                remove => SystemEvents.SessionSwitch -= remove);
```